src/mudlet.cpp

```diff
@@ -2307,8 +2357,13 @@ void mudlet::writeSettings()
     settings.setValue("size", size());
     settings.setValue("mainiconsize", mToolbarIconSize);
     settings.setValue("tefoldericonsize", mEditorTreeWidgetIconSize);
-    settings.setValue("showMenuBar", mShowMenuBar);
-    settings.setValue("showToolbar", mShowToolbar);
+    // This pair are only for backwards compatibility and will be ignored for
+    // this and future Mudlet versions - suggest they get removed in Mudlet 4.x
+    settings.setValue("showMenuBar", mMenuBarVisibility != visibleNever);
+    settings.setValue("showToolbar", mToolbarVisibility != visibleNever);
+
+    settings.setValue("menuBarVisibility", static_cast<int>(mMenuBarVisibility));
+    settings.setValue("toolBarVisibility", static_cast<int>(mToolbarVisibility));
```

Comment:
What is the point of renaming the options and breaking old settings?

Reply:
The old setting was a boolean and that cannot handle _three_ states.

Reply:
FTR It will leave _compatible_ settings behind for current versions of Mudlet but will only use the new ones.